custom-instrumentation/instrument/index.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const newrelic = require('newrelic')
+const Queue = require('./job-queue')
+
+function exampleJob() {
+    // Do whatever work you want here - this is just a placeholder
+    return 'job done';
+}
+
+function cbJob(cb) {
+    const result = exampleJob();
+    return cb('Callback ' + result);
+}
+
+async function promiseJob() {
+    return new Promise((resolve, reject) => {
+        try {
+            const result = exampleJob()
+            resolve('Promise ' + result)
+        }
+        catch (error) {
+            reject(error)
+        }
+    })
+}
+
+function main(){
+    const queue = new Queue()
+    newrelic.startBackgroundTransaction('firstTransaction', function first() {
```

Comment:
I think we should explain why we are starting a transaction here. I suspect more than a few people will think that `newrelic.instrument` in the `instrumentation.js` file is all that is required and will be confused by this API usage.

Basically, we should explain that since this application is not utilizing existing instrumented frameworks, a transaction is not automatically created that would encapsulate the instrumentation done in the other file. So we need to manually start the transaction for the instrumentation to be picked up.